src/Visualizer/Visualizer.js

```diff
@@ -32,21 +40,22 @@ class Visualizer {
   change_node_position(new_position) {
     let Node = new_position.node;
     Node.position = new_position.position;
+    new_position.animate();
   }
-
+  // new node color will by only chaned in the Graph  and won't be animated
   change_node_color(new_color) {
     let Node = new_color.node;
     Node.color = new_color.color;
   }
 
   change_node_size(new_size) {
     let Node = new_size.node;
-    Node.size = new_size.size
-
+    Node.size = new_size.size;
+    new_size.animate();
   }
-
+  // new edge color will by only chaned in the Graph  and won't be animated (as it doesn't make sense).
```

Comment:
the change of the color should be animated on the canvas...we already have an animation for it